platform/security/idp/security-idp-plugin/security-idp-plugin-subjectconfirmation/src/main/java/org/codice/ddf/security/idp/plugin/subjectconfirmation/SubjectConfirmationPresignPlugin.java

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.security.idp.plugin.subjectconfirmation;
+
+import static java.util.Objects.nonNull;
+
+import com.google.common.collect.Maps;
+import ddf.security.samlp.MetadataConfigurationParser;
+import ddf.security.samlp.SamlProtocol.Binding;
+import ddf.security.samlp.impl.EntityInformation;
+import ddf.security.samlp.impl.EntityInformation.ServiceInfo;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+import org.codice.ddf.security.idp.plugin.SamlPresignPlugin;
+import org.joda.time.DateTime;
+import org.opensaml.saml.saml2.core.Assertion;
+import org.opensaml.saml.saml2.core.AuthnRequest;
+import org.opensaml.saml.saml2.core.Response;
+import org.opensaml.saml.saml2.core.SubjectConfirmation;
+import org.opensaml.saml.saml2.core.SubjectConfirmationData;
+import org.opensaml.saml.saml2.core.impl.SubjectConfirmationDataBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SubjectConfirmationPresignPlugin implements SamlPresignPlugin {
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(SubjectConfirmationPresignPlugin.class);
+
+  @Override
+  public void processPresign(
+      Response response,
+      AuthnRequest authnRequest,
+      List<String> spMetadata,
+      Set<Binding> supportedBindings) {
+
+    String inResponseTo = response.getInResponseTo();
+
+    final DateTime notOnOrAfter = DateTime.now().plusMinutes(30);
```

Comment:
❓  Curious why this variable gets to be final, but the others nearby do not.

Reply:
Honestly...no particular reason. It does not _need_ to be final, nor do those not final around it need to be mutable. I can change for consistency's sake if it's bothersome enough.